### ISSUE TITLE
What is the difference between security levels “User”, “Staff” and “Superuser” Using Tendenci 11.0.2?

### ISSUE THREAD

**Issue by lashirley21** [tendenci/tendenci#787]
User question [cwalkden] : 
What is the difference between security levels “User”, “Staff” and “Superuser”? I have discovered by trial and error that I need to make someone “Staff” before they can run a report. I'd previously assumed that I could give this privilege to a group. I’d really like have it spelled out exactly what each level allows. Is this documented somewhere? Using Tendenci 11.0.2

**Comment by goetzk** [comment]
Question source is https://www.tendenci.com/forums/c/support/support/security-levels/

While not a complete answer the following may provide some more information.

'staff' status (specifically checks using user.is_staff) are used extensively in the forums module. 

There are also checks for staff via Django's staff_member_required in a many places:
* Several reports
* Corporate membership views
* Invoicing/recurring payments
* Memberships
* Articles
* and others...

user.is_admin is only used in half a dozen files.

Superuser/is_superuser is used all over the place with overlapping but not identical places to staff_member_required.

In summary, there doesn't appear to be a logical split between the two and its likely if you're giving someone staff permissions you'll need to give them admin/superuser too.

**Comment by eschipul** [comment]
@lashirley21 @cwalkden @goetzk 

@lashirley21 Karl's answer correctly describes the current situation. The origin is checking group permissions on every object on a page is too slow. One of my rules is that "a user is a user is a user" and everything after that is an attribute of that user.

It is much faster to say "if user.is_member do x" than it is to check a cascading series of security options. Rather than try to explain it here, I would point cwalkden to this help file:

[https://www.tendenci.com/help-files/interactive-user-guide/](https://www.tendenci.com/help-files/interactive-user-guide/)

The use of staff is more of an accident in that if we imported a module from a different open source project it might make use of it. Yes it should be consistent throughout the software, but given modules like forums are low volume spam traps (unfortunately) then it doesn't justify the time currently.

Again - this is a pretty good help file: [https://www.tendenci.com/help-files/interactive-user-guide/](https://www.tendenci.com/help-files/interactive-user-guide/)
